docs/js-guide/requirejs-to-webpack.rst

```diff
@@ -0,0 +1,214 @@
+==============================
+
+RequireJS `reached its end of life <https://github.com/requirejs/requirejs/issues/1816#issuecomment-707503323>`__
+in September of 2020. Since then, it has been increasingly difficult to work with as more modern libraries
+no longer provide AMD modules (the preferred module format of RequireJS), and use modern JavaScript
+syntax that is no longer compatible with RequireJS's build process. We decided to use Webpack as our
+replacement for RequireJS in September of 2024. We expect the migration to be brief and straightforward
+with the following challenges to highlight:
+
+1. The use of ``stripe`` will need to be updated to use the module from ``npm``, and we will need update
+    our front-end to use Stripe's credit card widgets library.
+2. Shimmed dependencies in requirejs that do no yet have an analogous ``exports-loader`` statement in
+    ``webpack.common.js`` will need a statement added and usage tested. This should be straightforward, but
+    there might be a couple challenging shims in this process.
+3. Tests will also need to be moved to using Webpack instead of RequireJS. The difference with webpack is that
+    the ``commcarehq`` or ``commcarehq_b3`` common module is always imported in the entry point. In RequireJS
+    this main module was included in ``hqwebapp/partials/requirejs.html``. However, Webpack can't reliably build
+    bundles with this global bundle separated from the entry point. This shouldn't cause issues for tests, but it might.
+
+
+Overview of the Process
+-----------------------
+
+Migrations should happen in pull-request "chunks", meaning that pull requests should contain migrations for
+all (or most) entry points within a single application or a set of closely related applications. Additionally,
+commits should be made separately for each entry point migration.
+
+If new ``exports-loader`` statements are added, it is recommended to test the changes on staging to ensure
+the functionality is maintained between production and staging.
+
+The following steps outline the migration process on a per entry point basis.
+
+As a reminder, Entry Points are modules that are included directly on a page using a bundler template tag,
+like the ones listed below. We want to migrate the RequireJS entry points to Webpack entry points:
+
+ - ``requirejs_main_b5`` to ``webpack_main``
+ - ``requirejs_main`` to ``webpack_main_b3``
+
+Entry points are the modules that the bundler (RequireJS or Webpack) uses to build a dependency graph so
+that it knows what bundle of javascript dependencies and page-specific code is needed to render that page.
+See `the Static Files Overview <https://github.com/dimagi/commcare-hq/blob/master/docs/js-guide/static-files.rst>`__
+for a more detailed explanation.
+
+Step 1: Identify the Entry Point and Details
+--------------------------------------------
+
+First, identify the entry point you would like to migrate. Additionally, we need to determine if the entry
```

Comment:
I think this whole step could be deleted, and instead just start the next step with "First, find either the requirejs_main or requirejs_main_b5 tag that references the entry point." There's no real reason you *can't* migrate a js module that's part of an ongoing bootstrap migration, right? You would just need to update both sets of js files.

Reply:
I was more thinking if someone was actively working on that template, not so much if the file was split. hmm, maybe i can re-word this

Reply:
(realize I'm jumping *right* on your comments)

I think the likelihood of both migrations happening at the same to the same file is really small.

Reply:
simplified the docs...hopefully it reads better